tests/test_helpers_files.py

```diff
@@ -0,0 +1,68 @@
+import chardet
+import codecs
+import tablib
+from django.test import TestCase
+from six import BytesIO, StringIO, text_type
+
+from multi_import.helpers import files
+from multi_import.exceptions import InvalidFileError
+
+
+class FileHelperTests(TestCase):
+
+    def test_decode_file_when_reading_utf_8(self):
+        file = BytesIO()
+        file.write('UTF-8 character: Đà'.encode('utf-8'))
+
+        file.seek(0)
+        file_contents = file.read()
+        charset = chardet.detect(file_contents)
+        encoding = charset['encoding']
+
+        self.assertEqual('utf-8', encoding)
+
+        decoded_file_contents = files.decode_contents(file_contents)
+        self.assertTrue(type(decoded_file_contents), str)
+
+    def test_decode_file_when_reading_utf_16(self):
+        file = BytesIO()
+        file.write('UTF-16 character: $'.encode('utf-16'))
+
+        file.seek(0)
+        file_contents = file.read()
+        charset = chardet.detect(file_contents)
+        encoding = charset['encoding']
+
+        self.assertEqual('UTF-16', encoding)
+
+        decoded_file_contents = files.decode_contents(file_contents)
+        self.assertTrue(type(decoded_file_contents), str)
+
+    def test_decode_file_when_reading_latin_1(self):
+        file = BytesIO()
+        file.write('Latin-1 character: ¥'.encode('ISO-8859-1'))
+
+        file.seek(0)
+        file_contents = file.read()
+        charset = chardet.detect(file_contents)
+        encoding = charset['encoding']
+
+        # Latin-1 and ISO-8859-1 are equivalent.
+        self.assertEqual('ISO-8859-1', encoding)
+
+        decoded_file_contents = files.decode_contents(file_contents)
+        self.assertTrue(type(decoded_file_contents), str)
+
+    def test_decode_file_when_reading_file_with_invalid_encoding(self):
+        file = BytesIO()
+        file.write('EUC-KR: 가나다라'.encode('EUC-KR'))
+
+        file.seek(0)
+        file_contents = file.read()
+        charset = chardet.detect(file_contents)
+        encoding = charset['encoding']
+
+        self.assertNotIn(encoding, ['utf-8', 'utf-16', 'ISO-8859-1'])
+
+        with self.assertRaises(InvalidFileError):
+            files.decode_contents(file_contents)
```

Comment:
Currently, when running tests,`InvalidFileError` is not raised. Many seem to be able to be decoded with 'ISO-8859-1'. tried with Chinese, Japanese, Russian, etc.. Will try a few more to find the test case.

Reply:
It seems that 'ISO-8859-1' will not throw `UnicodeDecodeError`; therefore, `InvalidFileError` won't be raised (Refs: http://python-notes.curiousefficiency.org/en/latest/python3/text_file_processing.html#files-in-an-ascii-compatible-encoding-best-effort-is-acceptable). After team discussion, we have decided to drop the test case when the error is raised.